tests/components/gree/test_climate.py

```diff
@@ -407,6 +414,58 @@ async def test_send_target_temperature(
         state.attributes.get(ATTR_CURRENT_TEMPERATURE)
         == fake_device.current_temperature
     )
+    assert state.state == HVAC_MODES.get(fake_device.mode)
```

Comment:
This adds a check that hvac mode didn't change in existing test for setting the temperature, while the new test checks that both temperature and hvac mode got changed.